src/plugins/data/common/opensearch_query/filters/range_filter.ts

```diff
@@ -124,19 +124,24 @@ export const buildRangeFilter = (
     filter.meta.formattedValue = formattedValue;
   }
 
-  params = mapValues(params, (value: any) => (field.type === 'number' ? parseFloat(value) : value));
+  params = mapValues(params, (value: any) =>
+    field.type === 'number' && typeof value !== 'bigint'
+      ? isFinite(value) && (value > Number.MAX_SAFE_INTEGER || value < Number.MIN_SAFE_INTEGER)
+        ? BigInt(value)
+        : parseFloat(value)
+      : value
```

Comment:
This type of ternary seems to pop up a few times. Have you considered pulling it out into a util function?

Reply:
Yeah. One was in core and one was in a plugin so I let it slide. If I see it again, I will pull it into a util.